tools/legacy_keyring.py

```diff
@@ -0,0 +1,142 @@
+import subprocess
+import sys
+from typing import List, Union
+
+import click
+from blspy import G1Element
+
+from chia.util.errors import KeychainUserNotFound
+from chia.util.keychain import KeyData, KeyDataSecrets, get_private_key_user
+from chia.util.misc import prompt_yes_no
+
+# Hacky way to make sure the keyring libs are there even if they are no longer part of chia dependencies.
+for dependency in ["keyrings.cryptfile==1.3.4", "keyring==23.6.0"]:
+    subprocess.call(["pip", "install", dependency], stdout=subprocess.DEVNULL)
+
+from keyring.backends.macOS import Keyring as MacKeyring  # noqa: E402
+from keyring.backends.Windows import WinVaultKeyring as WinKeyring  # noqa: E402
+from keyrings.cryptfile.cryptfile import CryptFileKeyring  # noqa: E402
+
+LegacyKeyring = Union[MacKeyring, WinKeyring, CryptFileKeyring]
+
+
+CURRENT_KEY_VERSION = "1.8"
+DEFAULT_USER = f"user-chia-{CURRENT_KEY_VERSION}"  # e.g. user-chia-1.8
+DEFAULT_SERVICE = f"chia-{DEFAULT_USER}"  # e.g. chia-user-chia-1.8
+MAX_KEYS = 100
+
+
+def create_legacy_keyring() -> LegacyKeyring:
+    if sys.platform == "darwin":
+        return MacKeyring()
+    elif sys.platform == "win32" or sys.platform == "cygwin":
+        return WinKeyring()
+    elif sys.platform == "linux":
+        keyring: CryptFileKeyring = CryptFileKeyring()
+        keyring.keyring_key = "your keyring password"
+        return keyring
+    sys.exit(f"platform '{sys.platform}' not supported.")
```

Comment:
Probably https://click.palletsprojects.com/en/7.x/exceptions/ instead.  Or just our own handling.  But hiding `sys.exit()` around is annoying for reuse and testing.